src/helpers/PortManager.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright 2024 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// This class is used for automatic port allocation for Camel Run commands using --console parameter
+export class PortManager {
+    private usedPorts: Set<number> = new Set();
+    private readonly minPort: number;
+    private readonly maxPort: number;
+
+    constructor(minPort: number, maxPort: number) {
+        this.minPort = minPort;
+        this.maxPort = maxPort;
+    }
+
+    allocatePort(): number | undefined {
+        for (let port = this.minPort; port <= this.maxPort; port++) {
+            if (!this.usedPorts.has(port)) {
+                this.usedPorts.add(port);
```

Comment:
this must be searched not only to port used and previously allocated in this extension but also to all ports on the system

Reply:
maybe with something like that https://www.npmjs.com/package/detect-port

Reply:
it is especially important in case there are several VS Code instance running in parallel and using the extension as they will use the same range of ports